operators/src/source/gdal_source.rs

```diff
@@ -254,6 +254,11 @@ where
         let rasterband: GdalRasterBand =
             dataset.rasterband(data_set_params.rasterband_channel as isize)?;
 
+        // get the no_data_value. TODO: move to dataset parameters? Keep in mind, that this can change for each time step.
```

Comment:
I think we should add No Data value to GdalParameters s.t. we have all the meta data in one place.

Reply:
yes but what happens if the no-data value changes over time?

Reply:
with "over time" you mean you have different no-data values for different geotiffs? Well, you get one `GdalDataSetParameters` for each file and the meta data provider has to know the value somehow. If there are situations where there is no way of knowing it other than reading the tiff meta data, then we maybe need to enhance the `GdalDataSetParameters` to have all parameters as Options, and if they are None they are read from the tiff meta data.

Reply:
that could work

Reply:
I'll let you decide if you implement it now or create a new task for it

Reply:
we need to add a gdal metadata reader. so we should do it as a new task. https://trello.com/c/uddqQmtx

Reply:
da3aee9